spec/controllers/citizens_controller_spec.rb

```diff
@@ -0,0 +1,105 @@
+require 'rails_helper'
+
+RSpec.describe CitizensController do
+  describe 'GET #index' do
+    let(:citizens) { create_list(:citizen, 5) }
+
+    before { get :index }
+
+    it 'validate status' do
+      expect(response).to  have_http_status(:success)
+    end
+
+    it 'validate render template' do
+      expect(response).to  render_template(:index)
+    end
+
+    it 'validate citizens list' do
+      expect(assigns(:citizens).filtered_citizens).to match_array(citizens)
+    end
+  end
+
+  describe 'GET #new' do
+    before { get :new }
+
+    it 'validate status' do
+      expect(response).to have_http_status(:success)
+    end
+
+    it 'validate render template' do
+      expect(response).to render_template(:new)
+    end
+
+    it 'validate new object' do
+      expect(assigns(:citizen)).to be_a_new(Citizen)
+    end
+  end
+
+  describe 'POST #create' do
+    let(:citizen_params) { attributes_for(:citizen, birth_date: '2001-06-06'.to_date) }
+    let(:address_params) { attributes_for(:address) }
+
+    context 'When citizen have valid params' do
+      it 'validate create object' do
+        expect{
+          post :create, params: { citizen: citizen_params.merge!(address_attributes: address_params) }
+        }.to change(Citizen, :count)
+      end
+
+      it 'validate status' do
```

Comment:
Sempre que tiver mais de um `expect` no mesmo arquivo, faz assim `context 'description', aggregate_failures: true do`

Isso vai garantir que a pilha de testes não interrompa o processamento caso a linha 52 falhe, com o `aggregate_failures`, a linha 53 e 54 tb será executada, e o processamento de toda a pilha de testes não será interrompida. Sem falar nos benefícios de performance que ele nos trás.

https://pt.slideshare.net/Railwaymen_org/speed-up-rspec-tests-part-1
https://makandracards.com/makandra/65467-rspec-how-to-aggregate-failures

Reply:
Adicionei isso aos testes, dei uma lida sobre, não conhecia essa.